app/controllers/admin/batches_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Admin::BatchesController < ApplicationController
+  before_action :set_admin_batch, only: [:show, :edit, :update, :destroy]
+
+  def new
+    @batch = Batch.new
+    @roast = Roast.find(params[:roast])
+    @batch.roast = Roast.find(params[:roast])
+  end
+
+  def create
+    @batch = Batch.new(admin_batch_params)
+    if @batch.save
+      redirect_to admin_roast_url(@batch.roast_id), notice: 'Batch was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @batch.update(admin_batch_params)
+      redirect_to admin_roast_path(@batch.roast_id), notice: 'Batch was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    roast = @batch.roast_id
+    @batch.destroy
+    redirect_to admin_roast_url(roast), notice: 'Batch was successfully destroyed.'
+  end
+
+  private
+  # Use callbacks to share common setup or constraints between actions.
+  def set_admin_batch
+    @batch = Batch.find(params[:id])
+  end
+
+
```

Comment:
There are some code styles that are seen as best practices.
It is quite extensive to read but if you follow those code style you are good for most ruby projects. https://github.com/bbatsov/ruby-style-guide